src/Stratis.Features.FederatedPeg/FederationGatewayFeature.cs

```diff
@@ -126,6 +130,15 @@ internal class FederationGatewayFeature : FullNodeFeature
 
         public override async Task InitializeAsync()
         {
+            if (this.federationManager?.CurrentFederationKey?.PubKey != null &&
```

Comment:
Federation manager might be not initialized here. I think it will be a good idea to ensure that poa feature already initialized the fed manager

Reply:
That's being checked. There won't be a non-null `CurrentFederationKey` if the federation manager is not initialized as that is where its being set.